src/Func.h

```diff
@@ -100,8 +110,15 @@ class Func : public Obj
 		}
 
 	// Add a new event handler to an existing function (event).
-	virtual void AddBody(detail::StmtPtr new_body, const std::vector<detail::IDPtr>& new_inits,
-	                     size_t new_frame_size, int priority = 0);
+	[[deprecated("Remove in v6.1 - Use function_ingredients version")]] virtual void
+	AddBody(detail::StmtPtr new_body, const std::vector<detail::IDPtr>& new_inits,
+	        size_t new_frame_size, int priority = 0);
+
+	// Add a new event handler to an existing function (event).
+	virtual void AddBody(const zeek::detail::function_ingredients* ingredients);
+
+	// Update body disabled_count according to groups.
+	void UpdateBodies();
```

Comment:
This is just called from the registry, right? Feels a bit confusing to have it part of the normal, public API then, because it's not clear if/when anybody would call it. Couple ideas: make it non-public and the registry a friend, or put the logic to update the counter directly into the registry (and probably likewise make it a friend). 

(I'm not completely sure but if we did the latter and also added a mapping from event group to body to the registry (as I'm suggesting elsewhere), we might not need `groups` set inside the body anymore.)

Reply:
I moved the logic to EventRegistry and made it a private friend of Func. Being part of the public API was a mishap.

> (I'm not completely sure but if we did the latter and also added a mapping from event group to body to the registry (as I'm suggesting elsewhere), we might not need groups set inside the body anymore.)

Still as is with the motivation given elsewhere around not being able to have stable pointers to body's.